### PR TITLE
Improve tesseract_kinematics test coverage

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
This PR improves the unit test coverage for the tesseract_kinematics package along fixes for issues found in the process.

### CODE REVIEW

File: tesseract_kinematics/src/opw/opw_inv_kin.cpp [hunk 1]
```diff
@@ -104,7 +104,19 @@ bool OPWInvKin::checkJoints(const Eigen::Ref<const Eigen::VectorXd>& vec) const
...
+    {
+      CONSOLE_BRIDGE_logDebug("Joint %s is out-of-range (%g < %g < %g)",
+                              joint_names_[static_cast<size_t>(i)].c_str(),
+                              limits_.joint_limits(i, 0),
+                              vec(i),
+                              limits_.joint_limits(i, 1));
+    }
+  }
+
+  return true;
```
Comment (mpowelson) [inline tesseract_kinematics/src/opw/opw_inv_kin.cpp:119]: Shouldn't this still return false if outside the limits?

**Review by mpowelson — COMMENTED** [review]
I haven't looked at the unit tests yet.

File: tesseract_kinematics/src/opw/opw_inv_kin.cpp (hunk 1 above)
Comment (Levi-Armstrong) [reply tesseract_kinematics/src/opw/opw_inv_kin.cpp:119]: A while back we had switch most of them to not return false if out side joint limits and just print debug information.

File: tesseract_kinematics/include/tesseract_kinematics/kdl/kdl_fwd_kin_tree.h [hunk 2]
```diff
@@ -49,7 +49,7 @@ namespace tesseract_kinematics
  * @brief ROS kinematics functions.
  *
  * Typically, just wrappers around the equivalent KDL calls.
- *
+ * @todo The tip link sould be provided in the init and update srdf to require a tip link
```
[the remark this answers is not in the document]
Comment (Levi-Armstrong) [reply tesseract_kinematics/include/tesseract_kinematics/kdl/kdl_fwd_kin_tree.h:52]: fixed

**Comment by Levi-Armstrong** [comment]
@mpowelson You good with the changes. Functionally it did not change much mostly added a lot of tests testing all of the different branches for failures and making sure they produced the desired results.

**Pull request merged**